apps/storybook/src/Table.stories.tsx

```diff
@@ -1407,27 +1336,31 @@ export const ControlledState: Story<Partial<TableProps>> = (args) => {
           <Checkbox
             key={index}
             label={data.name}
-            checked={expandedRows[index]}
+            checked={expandedRows.some((row) => row.name === data.name)}
             onChange={(e) => {
-              setExpandedRows((rowIds) => {
-                const expandedRowIds = { ...rowIds };
-                if (e.target.checked) {
-                  expandedRowIds[index] = true;
-                } else {
-                  delete expandedRowIds[index];
-                }
-                return expandedRowIds;
-              });
+              tableInstance.current?.toggleRowExpanded(
+                [data.name],
+                e.target.checked,
+              );
             }}
           />
         ))}
       </InputGroup>
       <Table
         columns={columns}
         emptyTableContent='No data.'
-        useControlledState={controlledState}
-        stateReducer={tableStateReducer}
+        stateReducer={useCallback((newState, action, prevState, instance) => {
+          tableInstance.current = instance;
+          return newState;
+        }, [])}
         isSelectable
+        onSelect={useCallback((selected) => {
+          setSelectedRows(selected ?? []);
+        }, [])}
+        onExpand={useCallback((expanded) => {
+          setExpandedRows(expanded);
+        }, [])}
+        getRowId={useCallback((rowData) => rowData.name, [])} // In real world use instance id
```

Comment:
Maybe add `id` to data so no one would think of using `name` as a row id and random bugs if there are same names.

Reply:
Thanks, done (https://github.com/iTwin/iTwinUI/pull/1008/files/772767f26cb6a7601f7f9c1fdd3554935a1edc9e..9fddc890ea2d90cf4c76cd218f6f8f5927ea3a80)